R/test-package.r

```diff
@@ -72,16 +80,9 @@ test_check <- function(package, filter = NULL, reporter = "summary") {
     stop("No tests found for ", package, call. = FALSE)
   }
 
-  reporter <- find_reporter(reporter)
-  env <- test_pkg_env(package)
-  with_top_env(env, {
-    df <- test_dir(test_path, reporter = reporter, env = env, filter = filter)
-  })
+  res <- .run_tests(package, test_path, filter, reporter)
```

Comment:
Then this does need an intermediate assignment (and the same in `test_package()`)